From 1b6eaf5589a14763452cbe53382cc699cdeca141 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Fri, 15 Nov 2019 11:36:14 +0100
Subject: [PATCH] configure.ac: add --enable-tests

Allow the user to disable tests through --disable-tests, this is
especially useful for example to disable gdk-pixbuf dependency

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status:
https://gitlab.freedesktop.org/spice/spice-common/merge_requests/6]
---
 Makefile.am  | 6 +++++-
 configure.ac | 7 +++++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/subprojects/spice-common/Makefile.am b/subprojects/spice-common/Makefile.am
index 5402499..00623a1 100644
--- a/subprojects/spice-common/Makefile.am
+++ b/subprojects/spice-common/Makefile.am
@@ -1,7 +1,11 @@
 NULL =
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = python_modules common tests docs
+SUBDIRS = python_modules common docs
+
+if ENABLE_TESTS
+SUBDIRS += tests
+endif
 
 EXTRA_DIST =				\
 	meson.build			\
diff --git a/subprojects/spice-common/configure.ac b/subprojects/spice-common/configure.ac
index 9d10287..2dba7c8 100644
--- a/subprojects/spice-common/configure.ac
+++ b/subprojects/spice-common/configure.ac
@@ -33,6 +33,13 @@ AC_SEARCH_LIBS(regcomp, [regex rx])
 SPICE_CHECK_SYSDEPS
 SPICE_EXTRA_CHECKS
 
+AC_ARG_ENABLE([tests],
+  AS_HELP_STRING([--enable-tests],
+                 [Enable tests @<:@default=yes@:>@]),
+  [],
+  enable_tests="yes")
+AM_CONDITIONAL(ENABLE_TESTS, test "x$enable_tests" = "xyes")
+
 AC_ARG_ENABLE([alignment-checks],
   AS_HELP_STRING([--enable-alignment-checks],
                  [Enable runtime checks for cast alignment @<:@default=no@:>@]),
-- 
2.23.0

